From foo@baz Wed Jun  3 14:47:21 PDT 2009
Date: Wed, 03 Jun 2009 14:47:21 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: udlfb: fix some sparse warnings.

There are others remaining due to the __iomem namespace of the
framebuffer data pointer.

Cc: Roberto De Ioris <roberto@unbit.it>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/udlfb/udlfb.c |    6 +++---
 drivers/staging/udlfb/udlfb.h |    8 ++++----
 2 files changed, 7 insertions(+), 7 deletions(-)

--- a/drivers/staging/udlfb/udlfb.c
+++ b/drivers/staging/udlfb/udlfb.c
@@ -422,7 +422,7 @@ copyarea(struct dlfb_data *dev_info, int
 	return 1;
 }
 
-void dlfb_copyarea(struct fb_info *info, const struct fb_copyarea *area)
+static void dlfb_copyarea(struct fb_info *info, const struct fb_copyarea *area)
 {
 
 	struct dlfb_data *dev = info->par;
@@ -434,7 +434,7 @@ void dlfb_copyarea(struct fb_info *info,
 
 }
 
-void dlfb_imageblit(struct fb_info *info, const struct fb_image *image)
+static void dlfb_imageblit(struct fb_info *info, const struct fb_image *image)
 {
 
 	int ret;
@@ -447,7 +447,7 @@ void dlfb_imageblit(struct fb_info *info
 	/* printk("IMAGE BLIT (2) %d %d %d %d DEPTH %d {%p} %d!!!\n", image->dx, image->dy, image->width, image->height, image->depth, dev->udev, ret); */
 }
 
-void dlfb_fillrect(struct fb_info *info, const struct fb_fillrect *region)
+static void dlfb_fillrect(struct fb_info *info, const struct fb_fillrect *region)
 {
 
 	unsigned char red, green, blue;
--- a/drivers/staging/udlfb/udlfb.h
+++ b/drivers/staging/udlfb/udlfb.h
@@ -40,7 +40,7 @@ struct dlfb_video_mode {
 	uint8_t unknown3[4];
 } __attribute__ ((__packed__));
 
-struct dlfb_video_mode dlfb_video_modes[MAX_VMODES];
+static struct dlfb_video_mode dlfb_video_modes[MAX_VMODES];
 
 static void dlfb_bulk_callback(struct urb *urb)
 {
@@ -66,7 +66,7 @@ static int dlfb_bulk_msg(struct dlfb_dat
 	return dev_info->tx_urb->actual_length;
 }
 
-void dlfb_init_modes(void)
+static void dlfb_init_modes(void)
 {
 	dlfb_video_modes[0].col = 0;
 	memcpy(&dlfb_video_modes[0].hclock, "\x20\x3C\x7A\xC9", 4);
@@ -105,7 +105,7 @@ void dlfb_init_modes(void)
 	memcpy(&dlfb_video_modes[3].unknown3, "\x04\x02\x1E\x5F", 4);
 }
 
-char *dlfb_set_register(char *bufptr, uint8_t reg, uint8_t val)
+static char *dlfb_set_register(char *bufptr, uint8_t reg, uint8_t val)
 {
 	*bufptr++ = 0xAF;
 	*bufptr++ = 0x20;
@@ -115,7 +115,7 @@ char *dlfb_set_register(char *bufptr, ui
 	return bufptr;
 }
 
-int dlfb_set_video_mode(struct dlfb_data *dev_info, int width, int height)
+static int dlfb_set_video_mode(struct dlfb_data *dev_info, int width, int height)
 {
 	int i, ret;
 	unsigned char j;
